cppfd/parallel_mesh.cpp

```diff
@@ -0,0 +1,89 @@
+#include "parallel_mesh.h"
+
+#include <iomanip>
+#include <iostream>
+#include <array>
+#include <map>
+#include <cmath> 
+
+ParallelMesh::ParallelMesh(uint64_t n_x, uint64_t n_y, double cell_size,
+			   uint16_t n_p, uint16_t n_q,
+			   double o_x, double o_y){
+
```

Comment:
I would suggest using initializers here like this, which has cleaner semantics:

```C++
ParallelMesh::ParallelMesh(uint64_t n_x, uint64_t n_y, double cell_size,
			   uint16_t n_p, uint16_t n_q,
			   double o_x, double o_y)
  : n_cells_x(n_x),
  ......
{
...
}
```

Reply:
Thanks @lifflander I did not know you could use initializer syntax *and at the same time* have an implementation of the main body of the constructor